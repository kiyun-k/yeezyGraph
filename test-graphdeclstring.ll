; ModuleID = 'MicroC'

%struct.QueueId = type { %struct.Node*, %struct.Node*, i32 }
%struct.Node = type { i8*, %struct.Node* }
%struct.List = type { %struct.ListNode*, i32 }
%struct.ListNode = type { i8*, %struct.ListNode* }
%struct.pqueue = type { %struct.node**, i32, i32 }
%struct.node = type { i8*, i8, %struct.map*, %struct.map*, i8* }
%struct.map = type { i32, i32, i8**, i32* }
%struct.graph = type { %struct.List.5*, i32 }
%struct.List.5 = type { %struct.ListNode.4*, i32 }
%struct.ListNode.4 = type { i8*, %struct.ListNode.4* }

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00"
@fmt1 = private unnamed_addr constant [4 x i8] c"%s\0A\00"
@fmt2 = private unnamed_addr constant [3 x i8] c"%d\00"
@fmt3 = private unnamed_addr constant [3 x i8] c"%s\00"
@fmt4 = private unnamed_addr constant [4 x i8] c"%f\0A\00"
@str = private unnamed_addr constant [4 x i8] c"abc\00"
@str5 = private unnamed_addr constant [4 x i8] c"def\00"
@str6 = private unnamed_addr constant [4 x i8] c"abc\00"

declare i32 @printf(i8*, ...)

declare %struct.QueueId* @initQueueId()

declare void @enqueue(%struct.QueueId*, i8*)

declare void @dequeue(%struct.QueueId*)

declare i8* @front(%struct.QueueId*)

declare i32 @q_size(%struct.QueueId*)

declare %struct.List* @l_init()

declare void @l_add(%struct.List*, i8*)

declare void @l_delete(%struct.List*, i32)

declare i8* @l_get(%struct.List*, i32)

declare i32 @l_size(%struct.List*)

declare void @print_list(%struct.List*)

declare %struct.pqueue* @pq_init()

declare void @pq_push(%struct.pqueue*, %struct.node*)

declare %struct.node* @pq_delete(%struct.pqueue*)

declare i32 @p_size(%struct.pqueue*)

declare %struct.node* @n_init(i8*)

declare void @set_data(%struct.node*, i8*)

declare i8* @get_name(%struct.node*)

declare i1 @get_visited(%struct.node*)

declare void @modify_visited(%struct.node*, i1)

declare i8* @get_data(%struct.node*)

declare %struct.List* @get_inNodes(%struct.node*)

declare %struct.List* @get_outNodes(%struct.node*)

declare %struct.graph* @g_init()

declare void @addNode(%struct.graph*, %struct.node*)

declare void @removeNode(%struct.graph*, %struct.node*)

declare void @addEdge(%struct.graph*, %struct.node*, %struct.node*, i32)

declare void @removeEdge(%struct.graph*, %struct.node*, %struct.node*)

declare void @printGraph(%struct.graph*)

declare i1 @isEmpty(%struct.graph*)

declare i32 @size(%struct.graph*)

declare i1 @contains(%struct.graph*, i8*)

declare %struct.node* @getNode(%struct.graph*, i8*)

declare i32 @getWeight(%struct.graph*, %struct.node*, %struct.node*)

define i32 @main() {
entry:
  %g1 = alloca %struct.graph*
  %x = alloca i32
  %a1 = alloca i8*
  %a2 = alloca i8*
  %a3 = alloca i8*
  %a4 = alloca i8*
  %y = alloca i32
  %n1 = alloca %struct.node*
  %n2 = alloca %struct.node*
  %l = alloca %struct.List*
  %init = call %struct.List* @l_init()
  store %struct.List* %init, %struct.List** %l
  %init1 = call %struct.graph* @g_init()
  store %struct.graph* %init1, %struct.graph** %g1
  store i8* getelementptr inbounds ([4 x i8]* @str, i32 0, i32 0), i8** %a1
  store i8* getelementptr inbounds ([4 x i8]* @str5, i32 0, i32 0), i8** %a2
  %char_n_val_pointer = load i8** %a1
  %init2 = call %struct.node* @n_init(i8* %char_n_val_pointer)
  %node_alloca = alloca %struct.node*
  store %struct.node* %init2, %struct.node** %node_alloca
  %graph_pointer = load %struct.graph** %g1
  %node_pointer = load %struct.node** %node_alloca
  call void @addNode(%struct.graph* %graph_pointer, %struct.node* %node_pointer)
  %char_n_val_pointer3 = load i8** %a2
  %init4 = call %struct.node* @n_init(i8* %char_n_val_pointer3)
  %node_alloca5 = alloca %struct.node*
  store %struct.node* %init4, %struct.node** %node_alloca5
  %graph_pointer6 = load %struct.graph** %g1
  %node_pointer7 = load %struct.node** %node_alloca5
  call void @addNode(%struct.graph* %graph_pointer6, %struct.node* %node_pointer7)
  %g18 = load %struct.graph** %g1
  %char_n_val_pointer9 = load i8** %a1
  %0 = call %struct.node* @getNode(%struct.graph* %g18, i8* %char_n_val_pointer9)
  store %struct.node* %0, %struct.node** %n1
  %n110 = load %struct.node** %n1
  %malloccall = tail call i8* @malloc(i32 ptrtoint (i1** getelementptr (i1** null, i32 1) to i32))
  %tmp = bitcast i8* %malloccall to i8**
  store i8* getelementptr inbounds ([4 x i8]* @str6, i32 0, i32 0), i8** %tmp
  %ptr = bitcast i8** %tmp to i8*
  call void @set_data(%struct.node* %n110, i8* %ptr)
  %n111 = load %struct.node** %n1
  %getData = call i8* @get_data(%struct.node* %n111)
  %d_ptr = bitcast i8* %getData to i8**
  %d_ptr12 = load i8** %d_ptr
  store i8* %d_ptr12, i8** %a3
  %a313 = load i8** %a3
  %printf = call i32 (i8*, ...)* @printf(i8* getelementptr inbounds ([4 x i8]* @fmt1, i32 0, i32 0), i8* %a313)
  %g114 = load %struct.graph** %g1
  %char_n_val_pointer15 = load i8** %a2
  %1 = call %struct.node* @getNode(%struct.graph* %g114, i8* %char_n_val_pointer15)
  store %struct.node* %1, %struct.node** %n2
  %g116 = load %struct.graph** %g1
  %e1_val_pointer = load %struct.node** %n1
  %e2_val_pointer = load %struct.node** %n2
  call void @addEdge(%struct.graph* %g116, %struct.node* %e1_val_pointer, %struct.node* %e2_val_pointer, i32 2)
  %g117 = load %struct.graph** %g1
  %n118 = load %struct.node** %n1
  %n219 = load %struct.node** %n2
  %getWeight = call i32 @getWeight(%struct.graph* %g117, %struct.node* %n118, %struct.node* %n219)
  store i32 %getWeight, i32* %x
  %x20 = load i32* %x
  %printf21 = call i32 (i8*, ...)* @printf(i8* getelementptr inbounds ([4 x i8]* @fmt, i32 0, i32 0), i32 %x20)
  %g122 = load %struct.graph** %g1
  call void @printGraph(%struct.graph* %g122)
  %n123 = load %struct.node** %n1
  %getOutNodes = call %struct.List* @get_outNodes(%struct.node* %n123)
  store %struct.List* %getOutNodes, %struct.List** %l
  %l24 = load %struct.List** %l
  call void @print_list(%struct.List* %l24)
  %l25 = load %struct.List** %l
  %2 = call i32 @l_size(%struct.List* %l25)
  store i32 %2, i32* %y
  %y26 = load i32* %y
  %printf27 = call i32 (i8*, ...)* @printf(i8* getelementptr inbounds ([4 x i8]* @fmt, i32 0, i32 0), i32 %y26)
  ret i32 0
}

declare noalias i8* @malloc(i32)
