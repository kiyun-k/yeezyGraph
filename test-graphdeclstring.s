	.text
	.file	"test-graphdeclstring.ll"
	.globl	main
	.align	16, 0x90
	.type	main,@function
main:                                   # @main
	.cfi_startproc
# BB#0:                                 # %entry
	pushl	%esi
.Ltmp0:
	.cfi_def_cfa_offset 8
	subl	$72, %esp
.Ltmp1:
	.cfi_def_cfa_offset 80
.Ltmp2:
	.cfi_offset %esi, -8
	calll	l_init
	movl	%eax, 32(%esp)
	calll	g_init
	movl	%eax, 68(%esp)
	movl	$.Lstr, 60(%esp)
	movl	$.Lstr5, 56(%esp)
	movl	60(%esp), %eax
	movl	%eax, (%esp)
	calll	n_init
	movl	%eax, 28(%esp)
	movl	68(%esp), %ecx
	movl	%eax, 4(%esp)
	movl	%ecx, (%esp)
	calll	addNode
	movl	56(%esp), %eax
	movl	%eax, (%esp)
	calll	n_init
	movl	%eax, 24(%esp)
	movl	68(%esp), %ecx
	movl	%eax, 4(%esp)
	movl	%ecx, (%esp)
	calll	addNode
	movl	68(%esp), %eax
	movl	60(%esp), %ecx
	movl	%ecx, 4(%esp)
	movl	%eax, (%esp)
	calll	getNode
	movl	%eax, %esi
	movl	%esi, 40(%esp)
	movl	$4, (%esp)
	calll	malloc
	movl	$.Lstr6, (%eax)
	movl	%eax, 4(%esp)
	movl	%esi, (%esp)
	calll	set_data
	movl	40(%esp), %eax
	movl	%eax, (%esp)
	calll	get_data
	movl	(%eax), %eax
	movl	%eax, 52(%esp)
	movl	%eax, 4(%esp)
	movl	$.Lfmt1, (%esp)
	calll	printf
	movl	68(%esp), %eax
	movl	56(%esp), %ecx
	movl	%ecx, 4(%esp)
	movl	%eax, (%esp)
	calll	getNode
	movl	%eax, 36(%esp)
	movl	68(%esp), %ecx
	movl	40(%esp), %edx
	movl	%eax, 8(%esp)
	movl	%edx, 4(%esp)
	movl	%ecx, (%esp)
	movl	$2, 12(%esp)
	calll	addEdge
	movl	68(%esp), %eax
	movl	40(%esp), %ecx
	movl	36(%esp), %edx
	movl	%edx, 8(%esp)
	movl	%ecx, 4(%esp)
	movl	%eax, (%esp)
	calll	getWeight
	movl	%eax, 64(%esp)
	movl	%eax, 4(%esp)
	movl	$.Lfmt, (%esp)
	calll	printf
	movl	68(%esp), %eax
	movl	%eax, (%esp)
	calll	printGraph
	movl	40(%esp), %eax
	movl	%eax, (%esp)
	calll	get_outNodes
	movl	%eax, 32(%esp)
	movl	%eax, (%esp)
	calll	print_list
	movl	32(%esp), %eax
	movl	%eax, (%esp)
	calll	l_size
	movl	%eax, 44(%esp)
	movl	%eax, 4(%esp)
	movl	$.Lfmt, (%esp)
	calll	printf
	xorl	%eax, %eax
	addl	$72, %esp
	popl	%esi
	retl
.Ltmp3:
	.size	main, .Ltmp3-main
	.cfi_endproc

	.type	.Lfmt,@object           # @fmt
	.section	.rodata.str1.1,"aMS",@progbits,1
.Lfmt:
	.asciz	"%d\n"
	.size	.Lfmt, 4

	.type	.Lfmt1,@object          # @fmt1
.Lfmt1:
	.asciz	"%s\n"
	.size	.Lfmt1, 4

	.type	.Lfmt2,@object          # @fmt2
.Lfmt2:
	.asciz	"%d"
	.size	.Lfmt2, 3

	.type	.Lfmt3,@object          # @fmt3
.Lfmt3:
	.asciz	"%s"
	.size	.Lfmt3, 3

	.type	.Lfmt4,@object          # @fmt4
.Lfmt4:
	.asciz	"%f\n"
	.size	.Lfmt4, 4

	.type	.Lstr,@object           # @str
.Lstr:
	.asciz	"abc"
	.size	.Lstr, 4

	.type	.Lstr5,@object          # @str5
.Lstr5:
	.asciz	"def"
	.size	.Lstr5, 4

	.type	.Lstr6,@object          # @str6
.Lstr6:
	.asciz	"abc"
	.size	.Lstr6, 4


	.section	".note.GNU-stack","",@progbits
