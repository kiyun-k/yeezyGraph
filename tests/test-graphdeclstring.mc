int main() {
	graph<string> g1;
	int x;
	string a1;
	string a2;
        string a3;
	string a4;
	int y;
	node<string> n1;
	node<string> n2;
	list<string> l;
	l = new list<string>();
	g1 = new graph<string>();
	a1 = "abc";
	a2 = "def";
	g1~+a1;
	g1~+a2;
	n1 = g1~_a1;
        n1.setData("abc");
        a3 = n1@data;
        prints(a3);
	n2 = g1~_a2;
	g1[2]->(n1,n2);
	x = g1.weight(n1, n2);
	print(x);
	g1.printGraph();
	l = n1@outNodes;
	l.printList();
	y = l.lsize();
	print(y);
	return 0;
	
}
