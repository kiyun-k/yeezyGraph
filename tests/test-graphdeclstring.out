abc
2
graph: 
node name:  abc
node visited:  0
node inNodes:  {}
node outNodes:  {(def, 2), }

node name:  def
node visited:  0
node inNodes:  {(abc, 2), }
node outNodes:  {}

{def, }
1
